test/utils/padder/padder.go

```diff
@@ -0,0 +1,202 @@
+
+	nrtv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/test/utils/fixture"
+	nrtutil "github.com/openshift-kni/numaresources-operator/test/utils/noderesourcetopologies"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+)
+
+// This package allows to control the amount of available untilAvailableReach under the nodes.
+// it's doing that by "padding" the nodes with guaranteed pods until node reaches the desired amount of available untilAvailableReach.
+//
+// Usage example:
+// pad := New(client, "test-ns")
+// pad.Nodes(2).UntilAvailableReach(corev1.ResourceCPU, "3").UntilAvailableReach(corev1.ResourceMemory, "100Mi").Pad()
+// or
+// rl := corev1.ResourceList {
+//		corev1.ResourceCPU: "10",
+//		corev1.ResourceMemory: "100Mi"
+//	}
+// pad.Nodes(2).UntilAvailableReachWithResourceList(rl).Pad()
+
+const PadderLabel = "nrop-test-pad-pod"
+
+type Padder struct {
+	// Client defines the API client to run CRUD operations, that will be used for testing
+	Client    client.Client
+	Namespace string
+	*padRequest
+}
+
+type padRequest struct {
+	nNodes              int
+	untilAvailableReach corev1.ResourceList
+}
+
+// New return new padder object
+// ns is where padding pods will get deployed
+func New(cli client.Client, ns string) (*Padder, error) {
+	padder := &Padder{
+		Namespace: ns,
+	}
+
+	padder.Client = cli
+	padder.padRequest = &padRequest{untilAvailableReach: make(corev1.ResourceList)}
+	return padder, nil
+}
+
+// Nodes number of nodes to pad
+func (p *Padder) Nodes(n int) *Padder {
+	p.padRequest.nNodes = n
+	return p
+}
+
+// UntilAvailableReach will pad pods into nodes until reach the desired available untilAvailableReach
+func (p *Padder) UntilAvailableReach(resName corev1.ResourceName, quantity string) *Padder {
+	quan := resource.MustParse(quantity)
+	p.untilAvailableReach[resName] = quan
+	return p
+}
+
+// UntilAvailableReachWithResourceList is like UntilAvailableReach but with a complete ResourceList as a parameter
+func (p *Padder) UntilAvailableReachWithResourceList(resources corev1.ResourceList) *Padder {
+	p.untilAvailableReach = resources
+	return p
+}
+
+// Pad will create pad pods in order to align the nodes
+// with the requested amount of available untilAvailableReach
+func (p *Padder) Pad() error {
+	if p.nNodes == 0 {
+		klog.Warningf("no nodes for padding were found. please specify at least one node")
+		return nil
+	}
+
+	nrtList, err := nrtutil.GetUpdated(p.Client, nrtv1alpha1.NodeResourceTopologyList{}, time.Second*10)
+	if err != nil {
+		return err
+	}
+
+	// create a list of only nodes that have single-numa-node policy
+	singleNumaNrt := nrtutil.FilterTopologyManagerPolicy(nrtList.Items, nrtv1alpha1.SingleNUMANodeContainerLevel)
+	singleNumaNrt = append(singleNumaNrt, nrtutil.FilterTopologyManagerPolicy(nrtList.Items, nrtv1alpha1.SingleNUMANodePodLevel)...)
```

Comment:
is this correct? it seems convoluted

Reply:
We want all nodes with a ```single-numa-node``` policy, we don't care about the scope policy.

Reply:
ok, let's try to simplify and/or make the code more explicit though.

Reply:
OK let me revaluate

Reply:
I hope now it's more explicit